docs/concepts/time-and-windows.rst

```diff
@@ -0,0 +1,118 @@
+.. _time-and-windows:
+
+Time and Windows
+################
+
+During the life cycle of a record, it experiences a number of events
+which occur at particular times. Three important times in a record's 
+life cycle are *Event-time*, *Ingestion-time*, and *Processing-time*. 
+
+Representing time consistently enables aggregation operations on streams,
+like SUM, that have time boundaries.
+
+KSQL supports the following notions of time:

```

Comment:
>  It can be the event-time, some concept of ingestion-time, (i.e. ingested into the platform at time x, which would be different to ingested into Kafka at time y), or something random).

That is correct. However, we should consider all of those as "event-time". From a KSQL point of view, it's unknown how the timestamp is set -- thus, for KSQL it is just event-time IMHO. If user sets event-time in a crazy manner, it's the user's fault.

Ingestion-time, would be an approximation of event-time, but it technically still event-time.

Processing-time would only hold, if KSQL uses the current wall-clock time when reading the record -- I think, this is not supported in KSQL, as users cannot set a custom `TimestampExtractor` from my understanding?

I think, it overall boils down to: if data is reprocessed, so we use the same timestamps. For event-time this is true (also for ingestion-time) -- for processing-time, this is not true.